// material-ui-youtube-autocomplete/lib/index.js
'use strict';

Object.defineProperty(exports, '__esModule', {
	value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _get = function get(_x, _x2, _x3) { var _again = true; _function: while (_again) { var object = _x, property = _x2, receiver = _x3; _again = false; if (object === null) object = Function.prototype; var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { _x = parent; _x2 = property; _x3 = receiver; _again = true; desc = parent = undefined; continue _function; } } else if ('value' in desc) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } } };

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _materialUi = require('material-ui');

var _jsonp = require('jsonp');

var _jsonp2 = _interopRequireDefault(_jsonp);

var _youtubeFinder = require('youtube-finder');

var _youtubeFinder2 = _interopRequireDefault(_youtubeFinder);

var googleAutoSuggestURL = '//suggestqueries.google.com/complete/search?client=youtube&ds=yt&q=';

var MaterialUIAutocomplete = (function (_Component) {
	_inherits(MaterialUIAutocomplete, _Component);

	function MaterialUIAutocomplete(props) {
		_classCallCheck(this, MaterialUIAutocomplete);

		_get(Object.getPrototypeOf(MaterialUIAutocomplete.prototype), 'constructor', this).call(this, props);
		this.onUpdateInput = this.onUpdateInput.bind(this);
		this.onNewRequest = this.onNewRequest.bind(this);
		this.YoutubeClient = _youtubeFinder2['default'].createClient({ key: this.props.apiKey });
		this.state = {
			dataSource: [],
			inputValue: ''
		};
	}

	_createClass(MaterialUIAutocomplete, [{
		key: 'performSearch',
		value: function performSearch() {
			var self = this,
			    url = googleAutoSuggestURL + this.state.inputValue;
			if (this.state.inputValue !== '') {
				(0, _jsonp2['default'])(url, function (error, data) {
					var searchResults, retrievedSearchTerms;

					if (error) return console.log(error);

					searchResults = data[1];

					retrievedSearchTerms = searchResults.map(function (result) {
						return result[0];
					});

					self.setState({
						dataSource: retrievedSearchTerms
					});
				});
			}
		}
	}, {
		key: 'onUpdateInput',
		value: function onUpdateInput(inputValue) {
			var self = this;
			this.setState({
				inputValue: inputValue
			}, function () {
				self.performSearch();
			});
		}
	}, {
		key: 'onNewRequest',
		value: function onNewRequest(searchTerm) {
			var self = this;
			//fixed search query not being added before search
			this.setState({inputValue : searchTerm })
			var params = {
				part: 'id,snippet',
				type: 'video',
				q: this.state.inputValue,
				maxResults: this.props.maxResults <= 50 ? this.props.maxResults : '50'
			};

			this.YoutubeClient.search(params, function (error, results) {
				if (error) return console.log(error);
				self.props.callback(results.items, searchTerm);
				self.setState({
					dataSource: []
				});
			});
		}
	}, {
		key: 'render',
		value: function render() {
			return _react2['default'].createElement(
				'div',
				null,
				_react2['default'].createElement(_materialUi.AutoComplete, {
					searchText: this.state.inputValue,
					floatingLabelText: this.props.placeHolder,
					filter: _materialUi.AutoComplete.noFilter,
					triggerUpdateOnFocus: true,
					dataSource: this.state.dataSource,
					onUpdateInput: this.onUpdateInput,
					onNewRequest: this.onNewRequest })
			);
		}
	}]);

	return MaterialUIAutocomplete;
})(_react.Component);

exports['default'] = MaterialUIAutocomplete;
module.exports = exports['default'];
